src/testUtils/integrationTest.ts

```diff
@@ -1,91 +1,194 @@
-import { readFileSync, writeFileSync } from 'fs';
+import * as fs from 'fs';
+import { writeFileSync } from 'fs';
 import * as fsExtra from 'fs-extra';
 import * as path from 'path';
+import { ICacheMeta } from '../cache/cache';
 import { EnvironmentType } from '../config/EnvironmentType';
 import { IPublicConfig } from '../config/IConfig';
 import { Context, createContext } from '../core/context';
 import { bundleDev, IBundleDevResponse } from '../development/bundleDev';
 import { env } from '../env';
-import { ensureDir, fastHash } from '../utils/utils';
+import { ensureDir, fastHash, fileExists, path2RegexPattern, readFile, writeFile } from '../utils/utils';
+export interface ITestBrowserResponse {
+  __fuse: any;
+  entry: any;
+  errors: Array<any>;
+}
+export function createDevSandbox(props: { ctx: Context; response: IBundleDevResponse; workspace: ITestWorkspace }) {
+  const self = {
+    runBundleInBrowser: async (): Promise<ITestBrowserResponse> => {
+      const app = readFile(path.join(props.workspace.distRoot, 'app.js'));
+      const window: any = {};
+      const __fuse: any = new Proxy(
+        {},
+        {
+          get: (obj, prop) => {
+            return window.__fuse[prop];
+          },
+        },
+      );
+      var fn = new Function('window', '__fuse', app);
+      const errors = [];
+      try {
+        fn(window, __fuse);
+      } catch (e) {
+        errors.push(e);
+      }
+      return new Promise((resolve, reject) => {
```

Comment:
you can omit the return and reject. 

`return new Promise(resolve => resolve({...})`

Reply:
Actually cleaner to have it in array, and jsom had a handler for that